File: GameStore.Infrastructure/Persistence/Videogames/Models/Discount.cs
using GameStore.Infrastructure.Persistence.Videogames.Enums;

namespace GameStore.Infrastructure.Persistence.Videogames.Models
{
    public class Discount
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public DiscountType Type { get; set; }
        public DiscountValueType ValueType { get; set; }

        public decimal Value { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }

        public bool IsActive { get; set; }

        // 🔗 Relaciones
        public ICollection<DiscountScope> DiscountScopes { get; set; }
            = new List<DiscountScope>();

        public Coupon? Coupon { get; set; }
    }
}

File: frontend/gamestore/src/types/discount/discount.ts
// ==============================
// Enums / Union Types
// ==============================
export type DiscountTargetType = "All" | "Videogame" | "Genre" | "Platform";
export type DiscountType = "Seasonal" | "Coupon";
export type DiscountValueType = "Percentage" | "Fixed";

// ==============================
// Scopes
// ==============================
export interface DiscountScopeBase {
  targetType: DiscountTargetType;
  targetId?: number;
}

export type CreateDiscountScopeRequest = DiscountScopeBase;

export type UpdateDiscountScopeRequest = DiscountScopeBase & {
  id?: string;
};

// ==============================
// Coupon
// ==============================
export interface CouponBase {
  code: string;
  maxUses?: number;
}

export type CreateCouponRequest = CouponBase;

export type UpdateCouponRequest = CouponBase & {
  id?: string;
};

// ==============================
// Discount Requests
// ==============================
export interface CreateDiscountRequest {
  name: string;
  type: DiscountType;
  valueType: DiscountValueType;
  value: number;
  startDate: string | null;
  endDate: string | null;
  isActive: boolean;
  discountScopes: CreateDiscountScopeRequest[];
  coupon?: CreateCouponRequest;
}

export interface UpdateDiscountRequest extends CreateDiscountRequest {
  discountScopes: UpdateDiscountScopeRequest[];
  coupon?: UpdateCouponRequest;
}

// ==============================
// List / Detail DTOs
// ==============================
export interface DiscountListItem {
  id: string;
  name: string;
  type: DiscountType;
  valueType: DiscountValueType;
  value: number;
  isActive: boolean;
  startDate: string;
  endDate: string;
}

export interface DiscountDetailDto {
  id: string;
  name: string;
  type: DiscountType;
  valueType: DiscountValueType;
  value: number;
  startDate: string;
  endDate: string;
  isActive: boolean;
  discountScopes: UpdateDiscountScopeRequest[];
  coupon?: UpdateCouponRequest;
}
